src/validation_workflow/api_test_cases.py

```diff
@@ -0,0 +1,51 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import time
+from typing import Any
+
+from validation_workflow.api_request import ApiTest
+from validation_workflow.validation_args import ValidationArgs
+
+'''
+This class is the collection of test cases to run.
+'''
+
+
+class ApiTestCases:
+
+    def __init__(self) -> None:
+        self.OS_image_version = ValidationArgs().stg_tag('OS').split(' ')[0]
+        self.OSD_image_version = ValidationArgs().stg_tag('OSD').split(' ')[0]
+
+    def test_cases(self) -> Any:
+        pass_counter, fail_counter = 0, 0
+
+        # the test case parameters are formated as ['<request_url>',<success_status_code>,'<validate_string(optional)>']
+        test_cases = [
+            ['https://localhost:9200/', 200, '"number" : "' + self.OS_image_version + '"'],
+            ['https://localhost:9200/_cat/plugins?v', 200, ''],
+            ['https://localhost:9200/_cat/health?v', 200, 'green'],
+            ['http://localhost:5601/api/status', 200, '"number":"' + self.OSD_image_version + '"']
+        ]
+
+        for test_case in test_cases:
+            request_url = test_case.__getitem__(0)
+            success_status_code = test_case.__getitem__(1)
+            validate_string = test_case.__getitem__(2)
+
+            status_code, response_text = ApiTest(str(request_url)).api_get()
+            logging.info(f"\nStatus_code ->{status_code} \nresponse_text ->{response_text}")
+            time.sleep(3)
+
+            if status_code == success_status_code and (not validate_string or validate_string in response_text):
+                pass_counter += 1
+            else:
+                fail_counter += 1
+
+        return (fail_counter == 0, "There are " + str(pass_counter) + "/" + str(pass_counter + fail_counter) + " test cases Pass")
```

Comment:
Avoid using `/` but replace with `os.path.join` or `os.path.sep` or similar, so it can be used outside of nix* env.

Reply:
This applied to any similar issues in this PR.

Reply:
Thanks for the comment, @peterzhuamazon, this is a return string to show how many cases pass over the total test cases, I guess it won't be matter per OS, correct?

Reply:
This is just a print statement not a path so we are good

Reply:
thanks, @gaiksaya

Reply:
I am not saying this string only, see my comment I am talking about if there is any path that is trying this, we should aviod, also why not use `f` string but using `+` here?

Reply:
Everywhere else os.path.join has been used https://github.com/opensearch-project/opensearch-build/pull/3038/files#diff-968cdbdd15c8ab34a3bc9aa41df188abc7e459e0fc56f6affc36be763b78b898R144

Using f string is a nit. I dont think PR should be blocked for a print statement.

Reply:
thanks, @peterzhuamazon and @gaiksaya, and Yes, I just double check all the places that I need to concatenate path and file, I have them by using the os.path.join.

Reply:
Yeah Peter, we have used os.path.join omitting usage of "/" in all the scenarios. And have encapusulated the direct url links in f-string